hudi-common/src/main/java/org/apache/hudi/common/table/log/AbstractHoodieLogRecordReader.java

```diff
@@ -361,15 +372,38 @@ private boolean isNewInstantBlock(HoodieLogBlock logBlock) {
    */
   private void processDataBlock(HoodieDataBlock dataBlock, Option<List<String>> keys) throws Exception {
     try (ClosableIterator<IndexedRecord> recordItr = dataBlock.getRecordItr(keys.orElse(Collections.emptyList()))) {
+      boolean headRecord = true;
```

Comment:
Hoodie data log block stores schema in its header. So, Instead of reading the schema from the first record, can you directly use the header schema ? 
You can look at HoodieDataBlock.getWriterSchema(super.getLogBlockHeader()) for some idea

Reply:
already fixed, thanks very much